lib/Frontend/CompilerInvocation.cpp

```diff
@@ -633,6 +643,27 @@ static bool ParseLangArgs(LangOptions &Opts, ArgList &Args,
                      A->getAsString(Args), A->getValue());
     }
   }
+  
+  // Check if the locale code is supported.
+  if (Arg *A = Args.getLastArg(OPT_locale)) {
+    std::string localeCode = A->getValue();
+    
+    auto result = llvm::find_if(supportedLocaleCodes, [&](const char * locale) {
+      return localeCode == locale;
+    });
+    
+    if (result == std::end(supportedLocaleCodes)) {
+      std::string validLocaleCodes = "";
+      for (auto code: supportedLocaleCodes) {
+        validLocaleCodes += code;
+        validLocaleCodes += ", ";
+      }
+      // Incorrect locale code shouldn't crash the compiler,
+      // and the compiler will fallback to the default messages.
+      Diags.diagnose(SourceLoc(), diag::error_invalid_locale_code,
+                     validLocaleCodes.substr(0, validLocaleCodes.size()-2));
```

Comment:
Once you have adopted `llvm::interleave` duping `substr` here could no longer be necessary.